### ISSUE TITLE
e2e test failures due to master being schedulable

### ISSUE THREAD

**Issue by justinsb** [kubernetes/kops#705]
I tried directly running the command for the failing test, and this is what I saw:

```
> kubectl exec --namespace=e2e-tests-pod-network-test-1nu3y -it host-test-container-pod -- /bin/sh -c "curl -q -s 'http://100.96.2.7:8080/dial?request=hostName&protocol=http&host=10.123.45.3&port=8080&tries=1'"

{"errors":["Get http://10.123.45.3:8080/hostName: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"]}
```

That suggested to me that the problem was that we can't cope with a 10.123.45.x IP, which is the master "fake IPs".

This would be fixed by https://github.com/kubernetes/kubernetes/pull/33347, but I can't actually see where a daemonset-style pod is being created... the master is not supposed to be schedulable...

cc @zmerlynn

**Comment by justinsb** [comment]
Spoke too soon...

`networking_utils.go` schedules on random nodes, as long as they are schedulable.

Another test that needs to be updated for the move to taints...

**Comment by zmerlynn** [comment]
Should be fixed by kubernetes/kubernetes#35937 (fingers crossed)
